### PR TITLE
Add test whether the current user can actually use docker.

### PR DESCRIPTION
**Pull request by abergmeier**
For optional docker repos, add fallback without calling docker at all.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by abergmeier** [comment]
@damienmg So this should fix the problem of trying to execute `docker`.

**Comment by abergmeier** [comment]
@damienmg Next problem after creating a dummy repo, is how to disable the tests, which then do no longer work.
I would need to disable all test of `//src/test/docker:test_cc_configure-`*.
Any hints how this could be done easily?

**Comment by abergmeier** [comment]
So I now have the docker tests disabled. The whole thing is quite ugly I am afraid.

File: src/test/docker/flavours.bzl [hunk 1]
```diff
@@ -35,3 +35,8 @@ def pull_images_for_docker_tests():
         dockerfile = "//src/test/docker:Dockerfile." + flavour,
         optional = True,
     )
+
+  docker_config(
+      name = "docker_config",
+      optional = True,
+  )
```
Comment (damienmg) [inline src/test/docker/flavours.bzl]: nit: ending new line

File: tools/build_defs/docker/utils.py [hunk 2]
```diff
@@ -45,12 +47,16 @@ def GetTarFile(f, name):
   Returns:
     The content of the file, or None if not found.
   """
-  with tarfile.open(f, 'r') as tar:
-    members = [tarinfo.name for tarinfo in tar.getmembers()]
-    for i in ['', './', '/']:
-      if i + name in members:
-        return tar.extractfile(i + name).read()
-    return None
+  with open(f, "rb") as file:
```
[the remark this answers is not in the document]
Comment (abergmeier) [reply tools/build_defs/docker/utils.py]: To ensure the path is in the exception error message, should the file e.g. be missing. `extractfile` does not guarantee that.

File: src/test/docker/docker_pull.bzl [hunk 3]
```diff
@@ -39,6 +63,18 @@ docker_build(
     visibility = ["//visibility:public"],
 )
 """)
```
[the remark this answers is not in the document]
Comment (abergmeier) [reply src/test/docker/docker_pull.bzl]: The problem is that once you are unable to locate docker/and are not allowed to use it, there is no good way of using `docker_build` anymore. The current code is nonsensical. I'll work on better handling.

**Comment by abergmeier** [comment]
I can split the PR into 2 if that makes you sleep lighter ;)

File: tools/build_defs/docker/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (damienmg) [reply tools/build_defs/docker/utils.py]: The thing is this is legacy docker_rules code. It definitely does not belong to that PR and we should probably switch to use rules_docker instead.

**Comment by abergmeier** [comment]
I know too little about _rules_docker_. Hope that this is all fixed once the new rules are getting used.

**Pull request closed without merging**